File: core/src/org/tetawex/ecf/util/RandomProvider.java
package org.tetawex.ecf.util;

import com.badlogic.gdx.math.RandomXS128;

/**
 * Created by Tetawex on 03.05.17.
 */
public class RandomProvider {
    private static RandomXS128 random;
    private static long seed=0;

    public static RandomXS128 getRandom() {
        if(random==null)
            random=new RandomXS128(seed);
        return random;
    }
    public static void setSeed(long newSeed){
        seed=newSeed;
        random=new RandomXS128(seed);
    }
}

File: core/src/org/tetawex/ecf/util/RandomProvider.kt
package org.tetawex.ecf.util

import com.badlogic.gdx.math.RandomXS128

/**
 * Created by Tetawex on 03.05.17.
 */
object RandomProvider {
    private var random: RandomXS128? = null
    private var seed: Long = 0

    fun getRandom(): RandomXS128 {
        if (random == null)
            random = RandomXS128(seed)
        return random!!
    }

    fun setSeed(newSeed: Long) {
        seed = newSeed
        random = RandomXS128(seed)
    }
}
